packages/flutter_tools/lib/src/html_utils.dart

```diff
@@ -62,6 +62,7 @@ class IndexHtml {
   void applySubstitutions({
     required String baseHref,
     required String? serviceWorkerVersion,
+    String? buildConfig,
```

Comment:
That's a good idea. One of the reasons I didn't do this right away is because actually passing the `serviceWorkerVersion` into `loadEntrypoint` is currently the way to "opt-in" to the service worker in general. So if we have it implicitly injected by the build config, it sort of changes the whole semantics of the startup. We actually could change these semantics, and maybe have some other way of opting into the service worker that's a bit more obvious, but it would require more changes to the engine/flutter.js code, and this was already getting a bit unwieldy and out of scope.

As for baseHref, it can't *just* be in the build config, because we need to substitute it in the `<base>` tag anyway. And since we're substituting the `<base>` tag anyway and flutter.js just finds the `<base>` tag, making it part of the build config is actually superfluous. Maybe it would be better to explicitly pass it in the build config as well, but I'd have to change the engine/flutter.js code in order to do that.

Reply:
SGTM! Please, do create an issue with any improvements that you can think of that aren't happening now, so we don't forget about them (or you don't have to remember them forever :P)